components/mission_key_management/client.c

```diff
@@ -30,26 +69,57 @@ uint32_t client_state_epoll_events(enum client_state state) {
         case CS_DONE:
             return 0;
 
-        default:
-            // Unreachable
+        default: { 
+            /*$ assert(false); $*/  // <-- Prove this state is unreachable
             return 0;
+        } 
     }
 }
 
-struct client* client_new(int fd) {
+/*$ 
+datatype OptionClientObject {
+    SomeClientObject {struct client obj}, 
+    NoneClientObject {}
+}
+predicate (OptionClientObject) OptionClientNew(pointer p, i32 fd, i32 state)
```

Comment:
Is it useful to have the predicates in a separate file?

Reply:
I'm not sure. It's very closely tied to the behavior of the function. It could certainly go in `client.h` instead?

Reply:
Lets make a recommendation for *best practices* - either have the predicates in an existing header file (if manageable, e.g. `client.h`) or a new header, e.g. `client_predicates.h`

Does that sound reasonable?